frontend/src/hooks/useEmoji.tsx

```diff
@@ -1,28 +1,42 @@
-import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useCallback, useMemo, useState } from 'react';
 import { Connection } from '@vonage/client-sdk-video';
 import { throttle } from 'lodash';
-import useSessionContext from './useSessionContext';
 import { EMOJI_DISPLAY_DURATION } from '../utils/constants';
-
-type SignalEventType = {
-  type?: string;
-  data?: string;
-  from: Connection | null;
-};
+import { SignalEvent, SignalType, SubscriberWrapper } from '../types/session';
 
 type EmojiDataType = {
   emoji: string;
   time: number;
 };
 
+export type UseEmojiProps = {
+  signal: ((data: SignalType) => void) | undefined;
+  getConnectionId: () => string | undefined;
+};
+
+export type UseEmoji = {
+  sendEmoji: (emoji: string) => void;
+  emojiQueue: EmojiWrapper[];
+  onEmoji: (event: SignalEvent, subscriberWrappers: SubscriberWrapper[]) => void;
+};
+
 export type EmojiWrapper = {
   name: string;
   emoji: string;
   time: number;
 };
 
-const useEmoji = () => {
-  const { session, subscriberWrappers } = useSessionContext();
```

Comment:
We remove the session context as it's not necessary for this hook. Instead, we pass the `subscriberWrappers` directly. This also makes the tests simpler and easier to read.